lib/graphql/batch/execution_strategy.rb

```diff
@@ -49,7 +49,13 @@ def each_promise_in_entry(obj, key, value, &block)
     class FieldResolution < GraphQL::Query::SerialExecution::FieldResolution
       def get_finished_value(raw_value)
         if raw_value.is_a?(::Promise)
-          raw_value.then { |result| super(result) }
+          on_reject = lambda do |error|
+            raise error unless error.is_a?(GraphQL::ExecutionError)
+            error.ast_node = ast_node
+            query.context.errors << error
+            nil
+          end
+          raw_value.then { |result| super(result) }.then(nil, on_reject)
```

Comment:
:+1: to a `rescue` or `catch` method on the promise.

Note that once https://github.com/rmosolgo/graphql-ruby/pull/94 lands this block should return the error (instead of nil) so that the gem can handle errors in non-null fields appropriately.